### PR TITLE
Add Alazar controllers that avg over records

### PR DESCRIPTION
**Pull request by jenshnielsen**
These are needed to be able to use the buffers to loop over other parameters

### CODE REVIEW

File: pytopo/rf/alazar_acquisition.py [hunk 1]
```diff
@@ -247,7 +317,19 @@ def post_acquire(self):
         return super().post_acquire().mean(axis=1)
 
 
+class AvgRecIQCtl(AvgRecDemodCtl):
+
+    def data_shape(self):
+        shp = list(super().data_shape())
```
Comment (astafan8) [inline pytopo/rf/alazar_acquisition.py]: `shp` unused?

File: pytopo/rf/alazar_acquisition.py [hunk 2]
```diff
@@ -195,6 +195,39 @@ def post_acquire(self):
         return (data.astype(np.float32) / (2**self._nbits)) - 0.5
 
 
+class AvgRecCtl(BaseAcqCtl):
+
```
Comment (astafan8) [inline pytopo/rf/alazar_acquisition.py]: having no docstrings, i assume, is ok because everybody knows what's going on, right?

Comment (jenshnielsen) [reply pytopo/rf/alazar_acquisition.py]: Yes that seems to be the assumption this file is written under. As there are no docstrings in it at the moment

Comment (astafan8) [reply pytopo/rf/alazar_acquisition.py]: thank you very much for the docstrings!!!

File: pytopo/rf/alazar_acquisition.py [hunk 3]
```diff
@@ -232,22 +358,135 @@ def post_acquire(self):
...
+        Returns:
+             A tuple of the names of dimensions of the data returned
+        """
+        return ('buffers', 'channels')
+
+    def post_acquire(self):
+        """
+        Average data from super method over all periods.
+        """
+        return super().post_acquire().mean(axis=1)
```
Comment (jenshnielsen) [inline pytopo/rf/alazar_acquisition.py]: This depends on calling data_shape on the super class to set up the relevant cos/sin arrays. Not sure how that ever worked but it's definitly wrong. This is the same for the matching buffer avg controller AvgIQCtl

Comment (jenshnielsen) [reply pytopo/rf/alazar_acquisition.py]: I pushed a "fix" for this but I think the code should be restructured a bit so don't merge just yet.

**Pull request merged**